### PR TITLE
fix/auto-connect-calendar-3582

### LINKED ISSUE #3582: Auto-Connect Calendar
### Is your proposal related to a problem?

When a calendar is disconnected, users are sent an email for them to reconnect their calendar. 
65bc-356b-3153-03fbf301b653

Users then have to navigate to their cal account, to their integrations, and then reconnect the calendar. They also then have to reconnect each calendar that calendar was connected to - to the event types in advance settings. 
### Describe the solution you'd like

What we can do is a “Reconnect button” that takes you to the auth flow again instead of “Disconnect” and then “Connect” again. This should also automatically reconnect the calendars that were connected to the event types previously (currently when we have to disconnect and reconnect a calendar, that calendar gets removed from all event types it was connected to.)

It would be great to also have a flag notification next to event types that do not have a calendar connected to them. It's hard to know if a calendar is actually connected or not unless you navigate to the advance settings. This feels pretty important as well. 

I believe that if there's no calendar, then it automatically picks the "default calendar" - the default calendar should be listed in the advanced setting anyways as it's not as intuitive.

### PR DESCRIPTION
**Pull request by alannnc**
## What does this PR do?

- Removes label that says no connected calendar to an event type
- Shows new info Alert on event-types page when no destinationCalendar (default calendar) is found
- Adds onCascade: true to destinationCalendar model
- Now we display destinationCalendar (default) in event-type/$id page in select when no specific calendar is selected.

Fixes #3582 

https://www.loom.com/share/dbf7372d65ad4104baea2bcc46b78980

**Environment**: Staging(main branch) / Production

## Type of change

- [X] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

- [ ] Remove any calendar, go to event-types pages an alert should be showing up
- [ ] Enter event-type/$id page and when no destinationCalendar it's selected for that event type we should display user.destinationCalendar(default) as placeholder in Select component.

## Checklist

- I haven't added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: apps/web/components/DestinationCalendarSelector.tsx [hunk 1]
```diff
@@ -11,13 +12,15 @@ interface Props {
   hidePlaceholder?: boolean;
   /** The external Id of the connected calendar */
   value: string | undefined;
+  destinationCalendar?: DestinationCalendar | null;
```
Comment (alannnc) [inline apps/web/components/DestinationCalendarSelector.tsx:15]: DestinationCalendar its now being received as prop to show it as placeholder when no option is selected.

File: apps/web/pages/event-types/index.tsx [hunk 2]
```diff
@@ -607,6 +607,23 @@ const EventTypesPage = () => {
...
+                      You can connect your calendar from
+                      <a href="/apps/categories/calendar" className="underline">
+                        here
+                      </a>
+                      .
+                    </Trans>
+                  }
+                />
+              )}
+
```
Comment (alannnc) [inline apps/web/pages/event-types/index.tsx:626]: New alert when user.destinationCalendar is not found

File: apps/web/public/static/locales/en/common.json [hunk 3]
```diff
@@ -1028,13 +1028,15 @@
   "current_username": "Current username",
   "example_1": "Example 1",
   "example_2": "Example 2",
-  "additonal_input_label": "Additonal input label",
+  "additional_input_label": "Additional input label",
   "company_size": "Company size",
   "what_help_needed": "What do you need help with?",
   "variable_format": "Variable format",
-  "custom_input_as_variable_info": "Ignore all special characters of the additonal input label (use only letters and numbers), use uppercase for all letters and replace whitespaces with underscores.",
+  "custom_input_as_variable_info": "Ignore all special characters of the additional input label (use only letters and numbers), use uppercase for all letters and replace whitespaces with underscores.",
```
Comment (alannnc) [inline apps/web/public/static/locales/en/common.json:1035]: Typo fixed

File: packages/prisma/migrations/20220817201039_/migration.sql [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+ALTER TABLE "DestinationCalendar" DROP CONSTRAINT "DestinationCalendar_userId_fkey";
+
+-- AddForeignKey
+ALTER TABLE "DestinationCalendar" ADD CONSTRAINT "DestinationCalendar_userId_fkey" FOREIGN KEY ("userId") REFERENCES "users"("id") ON DELETE CASCADE ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "DestinationCalendar" ADD CONSTRAINT "DestinationCalendar_eventTypeId_fkey" FOREIGN KEY ("eventTypeId") REFERENCES "EventType"("id") ON DELETE CASCADE ON UPDATE CASCADE;
+
+-- AddForeignKey
+ALTER TABLE "DestinationCalendar" ADD CONSTRAINT "DestinationCalendar_credentialId_fkey" FOREIGN KEY ("credentialId") REFERENCES "Credential"("id") ON DELETE CASCADE ON UPDATE CASCADE;
```
Comment (alannnc) [inline packages/prisma/migrations/20220817201039_/migration.sql:17]: missing onCascade relationship actions

**Review by zomars — APPROVED** [review]

**Pull request merged**